sentry_sdk/integrations/pymongo.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import absolute_import
+
+from sentry_sdk import Hub
+from sentry_sdk.integrations import DidNotEnable, Integration
+from sentry_sdk.tracing import Span
+from sentry_sdk.utils import capture_internal_exceptions
+
+from sentry_sdk._types import MYPY
+
+try:
+    from pymongo import monitoring
+except ImportError:
+    raise DidNotEnable("Pymongo not installed")
+
+if MYPY:
+    from typing import Dict
+
+    from pymongo.monitoring import (
+        CommandFailedEvent,
+        CommandStartedEvent,
+        CommandSucceededEvent,
+    )
+
+
+class CommandTracer(monitoring.CommandListener):
+    def __init__(self):
+        self._ongoing_operations = {}  # type: Dict[int, Span]
+
+    def _operation_key(self, event):
+        # type: (CommandFailedEvent | CommandStartedEvent | CommandSucceededEvent) -> int
+        return event.request_id
+
+    def started(self, event):
+        # type: (CommandStartedEvent) -> None
+        hub = Hub.current
+        if hub.get_integration(PyMongoIntegration) is None:
+            return
+        with capture_internal_exceptions():
+            command = dict(event.command)
```

Comment:
There is a function called `sentry_sdk.hub._should_send_default_pii()` that should be used to guard personal identifiable information. (by default this function returns `False`, this behavior can be changed with this option: https://docs.sentry.io/platforms/python/configuration/options/#send-default-pii)

Right now if you do inserts or make queries, all the user data is sent to Sentry. This must not be the case unless `_should_send_default_pii()` returns `True`.

Can you please make sure that all the values in the command are replaced by `%s`? 
(Maybe there is a way to get the redacted/parameterized command from MongoDB?)

Reply:
There is no such option. 😞 Those aren't events sent to the database but objects crafted specifically to be used by monitoring components and there is no way from there to ensure access the original query object - which doesn't solve the problem either as it's a wrapper around the raw BSON data. There is no SQL-like convention of parameter replacement in mongo.
I don't think there is any clean way to only substitute user-provided data either, especially when you consider protocol changes between pymongo versions.

Basically I can do one of the two things:
* Only add `command` part when PII can be sent.
* Replace values in this dict by `%s` (recursively?).

TBH I'd rather do the first one to avoid performance penalty.

Reply:
I guess doing suggestion 2 would be cool. But I do not think we need recursion right now.
Just something like this should be ok:
```
>>> bla = {
...   "blub": "hello",
...   "foo": {
...     "bar": "hello2",
...   }
... }
>>> bla.keys()
dict_keys(['blub', 'foo'])
>>> for key in bla.keys():
...   bla[key] = "%s"
... 
>>> bla
{'blub': '%s', 'foo': '%s'}
```

If we find out that this is not enough data to debug queries (or just know what queries are run) we can change them in the future.

Reply:
I've implemented the first one for now.

Reply:
Ah. Missed your comment. Will do it like suggested then.

Reply:
Added top level replacement.

Reply:
Oh great. Missed tests update.

BTW should I add tests for that PII part?

Reply:
Tests for PII stuff would be great, yes!

Reply:
Added.